### DISCUSSION TITLE
hi guys , i can't use the experimental generate commit feature after the new update

### DISCUSSION THREAD

**Discussion by Yassin-Samir** [gitkraken/vscode-gitlens#3140] [Q&A]
hi guys , i can't use the experimental generate commit feature after the new update. the icon has disappeared

**Comment by jkelroy** [comment]
Hello @Yassin-Samir, in 1.86 VS Code removed the ability for an extension to provide the icon in the commit box. However, they will be introducing the ability back in a later release.

For now, you can still use the command palette (command/ctrl + shift + P) to generate an AI commit message.

<img width="596" alt="Screenshot 2024-02-09 at 3 18 41 PM" src="https://github.com/gitkraken/vscode-gitlens/assets/27739784/093e116a-2e46-427f-8d02-1186698c3ca3">

**Reply by Yassin-Samir** [reply]
I can't find it

**Reply by Yassin-Samir** [reply]
![Screenshot (390)](https://github.com/gitkraken/vscode-gitlens/assets/49534950/f0e6204c-9a9a-47a4-b571-4cf86448c1b7)

**Reply by Yassin-Samir** [reply]
@jkelroy

**Reply by jkelroy** [reply]
Try opening the command palette (ctrl + shift + P), searching _Preferences: Open User Settings (JSON)_, and then adding/saving `"gitlens.ai.experimental.generateCommitMessage.enabled": true`. After that, do you see the option?

**Reply by jkelroy** [reply]
@JackCasica @Yassin-Samir we have found an issue that AI commit generation is not showing when you are not logged into GitLens. This has been addressed in the pre-release. You can switch to the pre-release edition of GitLens, by clicking on the "Switch to Pre-Release version of this extension" from the Extensions view.

![image](https://user-images.githubusercontent.com/27739784/240437025-f9f6b0cf-cda3-4b53-973b-3689f1e4481a.png)

**Comment by JackCasica** [comment]
It no longer shows up for me either in the command palette: 
<img width="1724" alt="image" src="https://github.com/gitkraken/vscode-gitlens/assets/76270077/dc342173-5b8c-4f6e-af76-dc9dd3804cf2">

However, I can still set and use a keybinding to the command: 
<img width="707" alt="image" src="https://github.com/gitkraken/vscode-gitlens/assets/76270077/8240456d-b72b-4dcf-b501-bf0dae293496">
